src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -98,7 +102,27 @@ public int getHttpStatus() {
   }
 
   public boolean isSuccess() {
-    return successCodes.contains(getHttpStatus());
+    boolean isSuccess = false;
+    // ServiceNow Rest API may return 200 OK response with an error message in the response body.
+    // Normally we would expect non-200 response code if there's an error.
+    if (checkRetryable()) {
+      isSuccess = false;
+    } else if (successCodes.contains(getHttpStatus())) {
+      isSuccess = true;
+    }
+    return isSuccess;
+  }
+
+  private boolean checkRetryable() {
+    Gson gson = new Gson();
+    JsonObject jo = gson.fromJson(this.getResponseBody(), JsonObject.class);
```

Comment:
getRetryable() will return false incorrectly (for retryble error) if isSuccess() was not called previously on the same RestAPIResponse. We shouldn't have such dependency between getters. You can fix this by moving the checkRetryable logic into the constructor.

Reply:
Moved the `checkRetryable()` call to the constructor and removed the dependency. https://github.com/data-integrations/servicenow-plugins/pull/16/commits/94cb7501d6219ca8756a6122b6daa510c0b7c1b8